daemon/api.go

```diff
@@ -303,16 +305,32 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 				resource = url.String()
 			}
 
-			results[name+"."+origin] = webify(m, resource)
+			fullname := name + "." + origin
+
+			// strings.Contains(fullname, "") is true
+			if !strings.Contains(fullname, searchTerm) {
+				continue
+			}
+
+			results[fullname] = webify(m, resource)
 		}
 	}
 
 	if includeStore {
-		// TODO: If there are no results (local or remote), report the error. If
-		//       there are results at all, inform that the result is partial.
-		found, _ := newRemoteRepo().All()
-		if len(found) > 0 {
+		repo := newRemoteRepo()
+		var found []snappy.Part
+
+		if searchTerm != "" {
+			found, _ = repo.Find(searchTerm)
 			sources = append(sources, "store")
+		} else {
+			found, _ = repo.All()
+
+			// TODO: If there are no results (local or remote), report the error. If
```

Comment:
This TODO was about the error ignored above. Can we please hold it above the offending line so we spot what this is about more easily?